packages/ui-modal/src/Modal/index.tsx

```diff
@@ -160,23 +149,14 @@ class Modal extends Component<ModalProps, ModalState> {
   renderChildren() {
     const { children, variant, overflow } = this.props
 
-    return Children.map(
-      children as (HeaderChild | BodyChild | FooterChild)[],
-      (child) => {
-        if (!child) return // ignore null, falsy children
-
-        if (matchComponentTypes<BodyChild>(child, [ModalBody])) {
-          return safeCloneElement(child, {
-            variant: variant,
-            overflow: child.props.overflow || overflow
-          })
-        } else {
-          return safeCloneElement(child, {
-            variant: variant
-          })
-        }
-      }
-    )
+    return Children.map(children, (child: any) => {
```

Comment:
Please o not use `any` if possible.  This line should be `return Children.map(children as ReactElement, (child) => {` because `safeCloneElement` expects a `ReactElement`. This makes it also necessary to add extra checks for non-React component children like a simple string. I think the best check is to use `isvalidElement` like here https://github.com/instructure/instructure-ui/blob/master/packages/ui-table/src/Table/index.tsx#L136